src/pybind/rados/rados.pyx

```diff
@@ -3031,6 +3038,34 @@ returned %d, but should return zero on success." % (self.name, ret))
         if ret < 0:
             raise make_ex(ret, "Ioctx.trunc(%s): failed to truncate %s" % (self.name, key))
         return ret
+   
+    @requires(('key', str_type), ('cmp_buf', bytes), ('offset', int))
+    def cmpext(self, key, cmp_buf, offset=0):
+        '''
+        Compare an on-disk object range with a buffer
+        :param key: the name of the object
+        :type key: str
+        :param cmp_buf: buffer containing bytes to be compared with object contents
+        :type cmp_buf: bytes 
+        :param offset: object byte offset at which to start the comparison
+        :type offset: int
+        
+        :raises: :class:`TypeError`
+        :raises: :class:`Error`
+        :returns: 0 - on success, negative error code on failure,
+                 (-MAX_ERRNO - mismatch_off) on mismatch
+        '''
+        self.require_ioctx_open()
+        key = cstr(key, 'key')
+        cdef:
+             char *_key = key
+             char *_cmp_buf = cmp_buf
+             size_t _cmp_buf_len = len(cmp_buf)
+             uint64_t _offset = offset
+        with nogil:
+            ret = rados_cmpext(self.io, _key, _cmp_buf, _cmp_buf_len, _offset)
+        assert ret < -MAX_ERRNO or ret == 0, "Ioctx.cmpext(%s): failed to compare %s" % (self.name, key)        
```

Comment:
can you please explain if ret is 0 then, why are we raising an error, since a 0 return value for success?

Reply:
if ret < -MAX_ERRNO or ret == 0，It doesn't throw an exception...

Reply:
thanks for explaining, lgtm!